src/main/java/com/google/api/codegen/config/LongRunningConfig.java

```diff
@@ -176,6 +182,57 @@ private static LongRunningConfig createLongRunningConfigFromProtoFile(
         totalPollTimeout);
   }
 
+  /** Creates an instance of LongRunningConfig from a Discovery method. */
+  @Nullable
+  static LongRunningConfig createLongRunningConfig(DiscoveryMethodModel methodModel) {
```

Comment:
There will be Discovery-specific additions to the LongRunningConfig later, but having a shared LongRunningConfig between Discovery and proto3 APIs makes it very easy to have the same generated output. This PR is much smaller than it would be if I had to insert a new DiscoLongRunningConfig into the pipeline and try to make it have the same features and surface as the proto3-based client libraries.

There are many instances in the generation pipeline that just call on `getReturnType` or `getMetadataType`. Also, we can just reuse the timeout settings from the proto3 configurations.

Reply:
Sorry, the dependency on discogapic package still seems unacceptable(L18). Also, this method is called `createLongRunningConfig`, which seems like a generic name (transport-independent), but accepts `DisoveryMethodModel` as an argument, which makes it discovery-only. I would be very easy to assume, looking at the method name and working on grpc gapics, that this method is supposed to be used there as well.

Reply:
Well it's only as unacceptable as the dependencies on proto3 entities (L20-25). Should everything be split off? Should the existing methods be renamed to reflect that they only apply for proto3 methods?

Reply:
This is how gapic-generator was designed - proto was the only protocol, so it was not hidden. After adding another protocol, yes, it should have been refactored first to introduce the proper abstraction for the protocol. This was done for `gax-java` for example, so now we have `gax`, `gax-grpc` and `gax-httpjson` where protocol is properly abstracted and the shared part (i.e. `gax`) can be used by both protocol-specific parts (`gax-grpc` and `gax-httpjson`). 

It wasn't done for gapic-generator. The world is not ideal, and probably it was a correct decision to not do it for gapic-genrator (not practical, as it would have been a complicated thing, with serious risk of breaking existing clients). 

My understanding of the current gapic-genrator architecture (in the protocol-specific aspect) was that gapic-generator did not try to isolate protobuf/grpc, but tries (as much as it is practically possible) to isolate discogapic. For example we have protocol-specific package  names, like `com.google.api.codegen.discogapic`, which is an "equivalent" of `gax-httpjson`, but do not have such thing for proto/grpc. Using the analogy with gax-java, I thought gapic-generator was architecturally like `gax` plus `gax-httpsjon`, where `gax` would have combined common and protobuf-specific parts.

As result, I thought it was acceptable to have protobuf stuff in common classes, but not acceptable to have httpjson (especially taking into account that proto/grpc is supported  by all 7 languages and httpsjon only by java).

If my assumptions (the above ones) are not correct, then there is not much we can do here, and mixing two protocols together becomes not optimal, but acceptable.